Sources/Datadog/Logger.swift

```diff
@@ -451,7 +461,8 @@ public class Logger {
             case (true, nil):
                 let logOutput = LogFileOutput(
                     fileWriter: loggingFeature.storage.writer,
-                    rumErrorsIntegration: LoggingWithRUMErrorsIntegration()
+                    rumErrorsIntegration: LoggingWithRUMErrorsIntegration(),
+                    reportingThreshold: self.datadogReportingThreshold
                 )
                 return (logBuilder, logOutput)
```

Comment:
I wonder if we shouldn't consider using [`ConditionalLogOutput`](https://github.com/DataDog/dd-sdk-ios/blob/a63954b0ca095d99205dd8000651527b447f03e6/Sources/Datadog/Logging/LogOutputs/LogUtilityOutputs.swift#L21-L22) instead of adding threshold information directly to `LogFileOutput`? It can be simpler approach - not requiring modifications to every instance of `LogFileOutput` in the SDK and only limited to this file.
```swift
let logOutput = ConditionalLogOutput(
   conditionedOutput: LogFileOutput(...), 
   condition: reportLogsAbove(threshold: datadogReportingThreshold)
)

func reportLogsAbove(threshold: LogLevel) -> (LogEvent) -> Bool {
    return { log in
        let logLevel = LogLevel(from: log.status)?.rawValue ?? 0
        return logLevel >= threshold.rawValue
    }
}
```
As a bonus, we could share `reportLogsAbove(threshold:)` logic with [internal logger](https://github.com/DataDog/dd-sdk-ios/blob/a63954b0ca095d99205dd8000651527b447f03e6/Sources/Datadog/Utils/InternalLoggers.swift#L65-L69) which does exactly the same conditioning.